pysma/__init__.py

```diff
@@ -302,3 +307,49 @@ def read(self, sensors):
             )
 
         return True
+
+
+    @asyncio.coroutine
+    def  read_logging(self, sensors, start_time, end_time):
+        """Read a logging key and return the results."""
+        if len(sensors) != 1:
+          _LOGGER.warning("logging sensor must be used alone")
+          return False
+
+        payload = { "destDev": [], "key": int(sensors[0].key), "tStart": start_time, "tEnd": end_time }
+        if self.sma_sid is None:
```

Comment:
Do you mind refactoring this a bit?

here or on line 260 create this function

```suggestion
    def read_body(self, url, payload):
        if self.sma_sid is None:
```

it should return `result_body` at the end and then both `read()` and `read_logging()` can use this new function that will log in, try to post and do some parsing/error messages...

Reply:
Done, make sure I have the  @asyncio.coroutine applied properly, I have no idea how they operate at this point of my education.

I ended up using the URL to determine how to handle the extension, it behaves as it did with results with keys (which these do not).  Having the function _read_body() return False or the result_body seemed strange and when I tried it I got a runtime error (RuntimeError: await wasn't used with future).

        notfound = []
        for sen in sensors:
            if sen.key in result_body:
                sen.extract_value(result_body)
                continue
            elif url == URL_LOGGER:
                sen.extract_logger(result_body)
                continue

            notfound.append(f"{sen.name} [{sen.key}]")

If you think having read() do the post-processing result_body is better than I will take a second look at why it failed for me.

Reply:
The reason it failed is because you called a async coroutine (`_read_body`) but didn't use the `await` keyword... 

So first off, there is an old syntax (used here) and a newer syntax for async functions / coroutines

```python
# OLD way
@asyncio.coroutine
def blocking_io():

# NEW way
async def blocking_io():
```

The way you call them is also different

```python
yield from blocking_io()  # old way
await blocking_io()  # new way
```


So adding `yield from _read body()` here should be the right way to call it so you can do postproccesing outside _read_body

Converting old to new syntax can be another possible hacktoberfest project!